### PR TITLE
Added check:spelling task

### PR DESCRIPTION
**Pull request by lslezak**
This moves https://github.com/yast/yast.github.io/pull/43 implementation here, to be shared across all YaST modules.

### CODE REVIEW

File: .spell.dict [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This is the repository specific custom dictionary
+#
+# Write the words in alphabetical order, one word per line.
+# If an unknown word is generic put it into the global dictionary
+# in lib/tasks/.spell.dict file.
```
Comment (mvidner) [inline .spell.dict]: IMHO hidden files should be used sparingly. It is usually OK as metadata in a top directory, but this is a regular data file so it should be named without the leading dot.

File: lib/tasks/spellcheck.rake [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#
+# The custom dictionaries contains one word per line.
+# The lines starting with '#' character are ignored (used for comments),
+#
+
+CUSTOM_DICTIONARY_FILE = ".spell.dict"
+
+# read the global and the repository custom dictionary
+def read_custom_words
+  # read the glodefault custom dictionary
```
Comment (mvidner) [inline lib/tasks/spellcheck.rake]: "glodefault"?

File: lib/tasks/spellcheck.rake [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    rescue LoadError
+      $stderr.puts "ERROR: Ruby gem \"raspell\" is not installed."
+      exit 1
+    end
+
+    # initialize aspell
+    speller = Aspell.new("en_US")
+    speller.suggestion_mode = Aspell::NORMAL
+    # ignore the HTML tags in the text
+    speller.set_option("mode", "html")
```
Comment (mvidner) [inline lib/tasks/spellcheck.rake]: The task is a bit long. I think factoring out the speller `require` and initialization would give just `speller = apell_speller` here.

File: lib/yast/rake.rb [hunk 4]
```diff
@@ -46,7 +46,7 @@ def self.spec_version
   conf.obs_project = "YaST:Head"
   conf.obs_sr_project = "openSUSE:Factory"
   conf.package_name = File.read("RPMNAME").strip if File.exists?("RPMNAME")
-  conf.version = Yast::Tasks.spec_version
+  conf.version = Yast::Tasks.spec_version if !Dir.glob("package/*.spec").empty?
```
Comment (mvidner) [inline lib/yast/rake.rb]: What is this for? Mention in .changes?

Comment (lslezak) [inline lib/yast/rake.rb]: Changes updated.

File: README.md [hunk 5]
```diff
@@ -20,14 +20,14 @@ Now lets check what it provides with `rake -T`
...
+Runs all RSpec tests ending with \_spec.rb or \_test.rb.
 
 ## run[client]
-Runs client with paths leading to git. Usefull to testing module without
+Runs client with paths leading to git. Useful to testing module without
 installation.
 
 ## console
-Runs ruby console with paths leading to git and Yast environment.
+Runs ruby console with paths leading to git and YaST environment.
```
Comment (jreidinger) [inline README.md]: new task is missing to be described and documented in README.md and also there should be noted, that local dictionary can be used

File: VERSION [hunk 6]
```diff
@@ -1 +1 @@
-0.1.10
+0.1.11
```
Comment (jreidinger) [inline VERSION]: it should be `0.2.0` as this library follow semantic versioning and this add functionality

File: lib/tasks/spellcheck.rake [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+CUSTOM_DICTIONARY_FILE = "spell.dict"
+
+# read the global and the repository custom dictionary
+def read_custom_words
+  # read the global default custom dictionary
+  dict_path = File.expand_path("../#{CUSTOM_DICTIONARY_FILE}", __FILE__)
+  custom_words = File.read(dict_path).split("\n")
+
+  if File.exist?(CUSTOM_DICTIONARY_FILE)
```
Comment (jreidinger) [inline lib/tasks/spellcheck.rake]: why not use `dict_path`? It is incosistent

File: lib/tasks/spellcheck.rake [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    puts "Loading custom dictionary (#{CUSTOM_DICTIONARY_FILE})..." if verbose == true
+    # read the custom dictionary from the project directory
+    custom_words += File.read(CUSTOM_DICTIONARY_FILE).split("\n")
+  end
+
+  # remove comments
+  custom_words.reject! { |word| word.start_with?("#")}
+  # remove duplicates
+  custom_words.uniq!
```
Comment (jreidinger) [inline lib/tasks/spellcheck.rake]: maybe report duplicates as warning?

File: lib/tasks/spellcheck.rake [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+  speller = Aspell.new("en_US")
+  speller.suggestion_mode = Aspell::NORMAL
+  # ignore the HTML tags in the text
+  speller.set_option("mode", "html")
+
+  speller
+end
+
+namespace :check do
+  desc "Run spell checker (by default for *.md and *.html files in Git)"
```
Comment (jreidinger) [inline lib/tasks/spellcheck.rake]: can we do it also for POT files?

File: VERSION (hunk 6 above)
Comment (lslezak) [inline VERSION]: Not necessarily, version `0.y.z` is for initial development and anything may change at any time, the API is considered unstable. See http://semver.org/ item "4.".

Comment (jreidinger) [inline VERSION]: still I prefer to use 0.2.0 so it is visible that it adds new functionality. I also hope that I can release 1.0 quickly, as I want to have stable API.

File: lib/tasks/spellcheck.rake (hunk 9 above)
Comment (lslezak) [inline lib/tasks/spellcheck.rake]: Um, POT file are little bit tricky - you need to generate it first, and it a has a specific format (gettext keywords and comments), we would have to preprocess the file.

Anyway, I'd rather add the spell check to rubocop-yast plugin (or even directly to upstream rubocop).

File: README.md (hunk 5 above)
Comment (lslezak) [inline README.md]: README update, thanks for noticing this!

File: README.md [hunk 10]
```diff
@@ -35,6 +35,20 @@ Collect translatable strings and create `*.pot` files.
 ## check:pot
 Check for common mistakes in translated texts.
 
+## check:spelling
+Checks spelling in `*.md` and `*.html` files. It uses a global custom dictionary
+(file `lib/tasks/spell.dict` in this repository) and supports repository specific
+dictionary (`spell.dict` file in the repository root directory).
```
Comment (jreidinger) [inline README.md]: is it `spell.dict` or `.spell.dict`? I think it is hidden file

File: README.md [hunk 11]
```diff
@@ -35,6 +35,20 @@ Collect translatable strings and create `*.pot` files.
...
+
+Note: The installed aspell English dictionary may differ in different products
```
Comment (jreidinger) [inline README.md]: can you do some emphasis on `Note` so it is easily visible? bold or cursive is enough.

File: README.md [hunk 12]
```diff
@@ -35,6 +35,20 @@ Collect translatable strings and create `*.pot` files.
...
+(esp. the Ubuntu dictionary used at Travis - a local check may pass, but it may
+later fail at Travis, be prepared for this...)
+
+Note: You need to explicitly install `aspell-devel`, `aspell-en`, `ruby-devel`
```
Comment (jreidinger) [inline README.md]: Same here, do some highlighting of `Note:` keyword.

File: README.md [hunk 13]
```diff
@@ -35,6 +35,20 @@ Collect translatable strings and create `*.pot` files.
...
+packages and the `raspell` Ruby gem. (The reason is to decrease the dependencies
+for the packages which do not use this task.)
```
Comment (jreidinger) [inline README.md]: can you place here example of zypper command for opensuse? like 
```
For opensuse run this command:
`sudo zypper in rubygem-raspell`
```

File: lib/tasks/spellcheck.rake (hunk 9 above)
Comment (jreidinger) [inline lib/tasks/spellcheck.rake]: ok.

File: README.md (hunk 10 above)
Comment (lslezak) [inline README.md]: I have changed that, see @mvidner's comment above.

Comment (jreidinger) [inline README.md]: I think @mvidner comment is about having `lib/tasks/spell.dict` ( not `lib/tasks/.spell.dict`) and `.spell.dict` in top level repository for specific stuff.

Comment (lslezak) [inline README.md]: I see, I misunderstood that, fixed.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**